greenpithumb/poller.py

```diff
@@ -256,9 +257,23 @@ def _poll_once(self):
 class _CameraPollWorker(_SensorPollWorkerBase):
     """Captures and stores pictures pictures from a camera."""
 
+    def __init__(self, scheduler, record_queue, camera_manager, light_sensor):
+        """Creates a new CameraPollWorker object.
+
+        Args:
+            scheduler: Poll time scheduler.
+            record_queue: Not used by CameraPollWorker.
+            camera_manager: Manages the capture and saving of photos.
+            light_sensor: An interface for reading the light level.
+        """
+        super(_CameraPollWorker, self).__init__(scheduler, record_queue,
+                                                camera_manager)
+        self._light_sensor = light_sensor
+
     def _poll_once(self):
         """Captures and stores an image."""
-        self._sensor.save_photo()
```

Comment:
Can we push more of the logic into `CameraManager`? Sort of mirroring the design of `PumpManager` with its moisture threshold and `pump_if_needed`.